Classes/Configuration/ConfigurationReader.php

```diff
@@ -190,15 +190,17 @@ protected function getConfigurationKeyByDomainDecode($configurationKey) {
 		$globalConfig = &$GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['realurl'];
 		foreach ($globalConfig['_DOMAINS']['decode'] as $domainName => $configuration) {
 			$checkThisConfiguration = false;
-			if ($domainName{0} == '/') {
-				// Regular expression, match only main host name
-				if (@preg_match($domainName, $this->hostName)) {
+			if($GLOBALS['TSFE']->beUserLogin < 1 || !$GLOBALS['TSFE']->beUserLogin){
```

Comment:
This condition does not                                               make sense: both mean that beUserLogin are false.